File: simulator/src/models/raw_data/parking_raw_data.py
from datetime import datetime
from uuid import UUID

from .raw_data import RawData


class ParkingRawData(RawData):

    def __init__(self, *, is_occupied: bool, latitude: float,
                 longitude: float, sensor_uuid: UUID, sensor_name: str,
                 timestamp: datetime = datetime.now(),
                 group_name: str | None = None) -> None:
        super().__init__(latitude=latitude, longitude=longitude, group_name=group_name,
                         sensor_uuid=sensor_uuid, sensor_name=sensor_name,
                         timestamp=timestamp)
        self.__is_occupied = is_occupied

    @property
    def topic(self) -> str:
        return "parking"

    def __eq__(self, other: any) -> bool:
        if not isinstance(other, ParkingRawData):
            return False
        return super().__eq__(other) and self.__is_occupied == other.__is_occupied

    def __hash__(self) -> int:
        return hash((super().__hash__(), self.__is_occupied))

File: flink/src/main/java/com/sevenlast/synccity/models/ParkingRawData.java
package com.sevenlast.synccity.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.apache.avro.generic.GenericRecord;

import java.time.LocalDateTime;
import java.util.Optional;
import java.util.UUID;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ParkingRawData extends RawData {
    private boolean isOccupied;

    public ParkingRawData(String sensorUuid, String sensorName, String groupName, double latitude, double longitude, LocalDateTime timestamp, boolean isOccupied) {
        super(sensorUuid, sensorName, groupName, latitude, longitude, timestamp);
        this.isOccupied = isOccupied;
    }

}
